dali/pipeline/operators/reader/reader_op.h

```diff
@@ -87,9 +88,9 @@ class DataReader : public Operator<Backend> {
       try {
         Prefetch();
       } catch (const std::exception& e) {
-        std::stringstream ss;
-        ss << "Prefetch Failed: " << e.what();
-        DALI_FAIL(ss.str());
+        prefetch_error_.reset(new std::string(e.what()));
+        consumer_.notify_all();
+        return;
```

Comment:
Shouldn't it also set prefetched_batch_ready_? In case nobody is waiting during the notify call.

Reply:
Yes it should.. I didn't save the file correctly
Fixed